### PR TITLE
MySQL User

### LINKED ISSUE #2: Provision MySQL Users
Implement support for provisioning MySQL users. In addition to provisioning databases on SQL servers users should also be supported. We'll start with the implementation for MySQL. Support for PostgreSQL can follow later.

## Implementation Details
* Create a new manged resource `User` in `mysql.sql.crossplane.io/v1alpha1`
* Implement a new `user` reconciler in the `pkg/controller/mysql/user` package
* Provision a user based on the CRD
  * The external name of the managed resource should be used as the user name
  * A password should be generated
  * The complete [ConnectionDetails](https://pkg.go.dev/github.com/crossplane/crossplane-runtime@v0.11.0/pkg/reconciler/managed#ConnectionDetails) should be returned

## Not in Scope
* User TLS options
* User password expire options
* User resource options
* User GRANTs (see #3)

## To be Discussed
I think a more ergonomic (i.e. typed) way to define GRANTs might make sense in the future. For now I'd like to start with a simple approach.

### PR DESCRIPTION
**Pull request by srueg**
### Description of your changes

Fixes #2 

Based on the refactoring in https://github.com/crossplane-contrib/provider-sql/pull/8

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested

Manually on a Kind cluster, in the unit tests and with an integration test.

`make test`
`make e2e.run`

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

File: apis/mysql/v1alpha1/databse_types.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
+/*
```
Comment (negz) [inline apis/mysql/v1alpha1/databse_types.go:1]: It looks like there's a typo in this filename - `databse` rather than `database`.

Comment (srueg) [reply apis/mysql/v1alpha1/databse_types.go]: Good catch, thanks 👍

**Comment by srueg** [comment]
The basics are implemented. Some refactoring was necessary of the existing code base, see #8.
What I don't quite understand is how the password could be updated/rotated? Should we add a secretRef to define a password instead of generating one and track its `resourceVersion`?

**Comment by negz** [comment]
> Should we add a secretRef to define a password instead of generating one and track its resourceVersion?

That sounds like a good idea; I believe we've started using that pattern in a few other places (RDS instances perhaps?).

**Review by negz — COMMENTED** [review]
Whoops - I started a review yesterday but forgot to submit. Looks like some of my comments might be updated now, but I'll push anyhow as I don't have time to review fully today.

File: pkg/clients/sql/postgresql.go [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+// IsDoesNotExist returns true if the supplied error indicates a database does
+// not exist.
+func (c PostgresDB) IsDoesNotExist(err error) bool {
+	if err == nil {
+		return false
+	}
+
+	// TODO(negz): Is there a less lame way to determine this?
+	return strings.HasSuffix(err.Error(), "does not exist")
+}
```
[the remark this answers is not in the document]
Comment (srueg) [reply pkg/clients/sql/postgresql.go:67]: Even better: I got rid of it altogether since the `IF EXISTS` query handles this already.

**Comment by srueg** [comment]
Thanks a lot for your review, really appreciate it! ✌️

**Review by negz — APPROVED** [review]
Thanks for working on this @srueg - it looks great! It may be worth updating the README either as part of this PR or a future one to document that we now support MySQL users.

**Comment by srueg** [comment]
Thanks again for the review! I'll give #3 a shot next.
Do you agree with the proposed design of the API?

**Pull request merged**